### PR TITLE
 Add attempt number to ObservedQuery

### PR DESCRIPTION
**Pull request by martin-sucha**
This will allow the observer to know whether the attempt was
original query or retry.

### CODE REVIEW

**Comment by martin-sucha** [comment]
Upstream PR for the first commit is at https://github.com/gocql/gocql/pull/1322, I'll create a separate upstream PR for the second commit when the first is merged.

**Comment by mrVanboy** [comment]
Would we like to have the same MR/PR policy as we have in our projects? One commit per merge request and rebasing only?

File: session.go [hunk 1]
```diff
@@ -662,6 +662,73 @@ type hostMetrics struct {
 type queryMetrics struct {
 	l sync.RWMutex
 	m map[string]*hostMetrics
+	// totalAttempts is total number of attempts.
+	// Equal to sum of all hostMetrics' Attempts.
+	totalAttempts int
+}
+
+// getHostMetricsLocked gets or creates host metrics for given host.
+func (qm *queryMetrics) getHostMetrics(host *HostInfo) *hostMetrics {
```
Comment (mrVanboy) [inline session.go]: I believe we could omit `get` prefix.

**Comment by rkuska** [comment]
> Would we like to have the same MR/PR policy as we have in our projects? One commit per merge request and rebasing only?

I don't think we need to as we have no CI in here and don't work on this so frequently. I would therefore adjust the workflow to upstream's (whichever project we decide to fork/contribute to).

File: session.go (hunk 1 above)
Comment (martin-sucha) [reply session.go]: Renamed.

**Comment by martin-sucha** [comment]
We already have this feature

**Pull request closed without merging**